velox/vector/fuzzer/VectorFuzzer.cpp

```diff
@@ -239,17 +239,46 @@ void fuzzFlatPrimitiveImpl(
   }
 }
 
+// Servers as a wrapper around a vector that will be used to load a lazyVector.
+// Ensures that the loaded vector will only contain valid rows for the row set
+// that it was loaded for. NOTE: If the vector is a multi-level dictionary, the
+// indices from all the dictionaries are combined.
+class VectorLoaderWrap : public VectorLoader {
+ public:
+  explicit VectorLoaderWrap(VectorPtr vector) : vector_(vector) {}
+
+  void loadInternal(RowSet rowSet, ValueHook* hook, VectorPtr* result)
+      override {
+    VELOX_CHECK(!hook, "VectorLoaderWrap doesn't support ValueHook");
+    SelectivityVector rows(rowSet.back() + 1, false);
+    for (auto row : rowSet) {
+      rows.setValid(row, true);
+    }
+    rows.updateBounds();
+    *result = makeEncodingPreservedCopy(rows);
+  }
+
+ private:
+  // Returns a copy of 'vector_' while retaining dictionary encoding if present.
+  // Multiple dictionary layers are collapsed into one.
```

Comment:
Why, because we need to decode it?

Reply:
For the latter, though this is the case now, it would be nice to ensure that the expression eval component is flexible enough to handle nested dictionary levels - say in case we extend the readers in the future to produce more flexible lazy/dictionary combinations.

Reply:
Good point, I will file a follow up task for this